paddle/fluid/framework/tensor.h

```diff
@@ -203,6 +181,28 @@ class Tensor {
    *          PlaceHolder::ptr_ and where the tensor data really begins.
    */
   size_t offset_;
+
+#ifdef PADDLE_WITH_MKLDNN
+
+ public:
+  inline mkldnn::memory::format format() const { return format_; }
+
+  inline void set_format(const mkldnn::memory::format format) {
+    format_ = format;
+  }
+
+ protected:
+  /**
+   * @brief the detail format of memory block which have layout as kMKLDNN
+   *
+   * @note MKLDNN lib support various memory format like nchw, nhwc, nChw8C,
+   *       nChw16c, etc. For a MKLDNN memory block, layout will be set as
+   *       DataLayout::kMKLDNN meanwhile detail memory format will be kept in
+   *       this field.
+   */
+
+  mkldnn::memory::format format_ = mkldnn::memory::format::format_undef;
+#endif
```

Comment:
why modify the `tensor.h`?

Reply:
There is s segmentation fault with the original tensor.h, as issue discussed: https://github.com/PaddlePaddle/Paddle/issues/19922

Reply:
@jczaja Could you help review it?

Reply:
@mapingshuo  This change in tensor.h should not be needed anymore. This was workaround to make demo working , before root cause was found. root cause was lack of -DPADDLE_WITH_MKLDNN . After this flag is added you should not modify tensor.h . Just adding a flag -DPADDLE_WITH_MKLDNN should be enough (It works on my setup). Please revert tensor.h changes and retest demo_trainer .

Reply:
Thank you, I understand now!

Reply:
> @mapingshuo This change in tensor.h should not be needed anymore. This was workaround to make demo working , before root cause was found. Root cause was lack of -DPADDLE_WITH_MKLDNN in CMakeLists.txt of demo ci . After this flag was added you should not modify tensor.h . Just adding a flag -DPADDLE_WITH_MKLDNN should be enough (It works on my setup). Please revert tensor.h changes and retest demo_trainer .

I have restored tensor.h, please review this PR again.

Reply:
@jczaja